File: chor-extraction/src/main/java/ast/sp/labels/Communication.java
package ast.sp.labels;

import ast.sp.interfaces.ExtractionLabel;
import ast.sp.nodes.expr.BooleanExpression;

public class Communication implements ExtractionLabel {
    private final String sender;
    private final String receiver;
    private final BooleanExpression expression;

    public Communication(String sender, String receiver, BooleanExpression expression) {
        this.sender = sender;
        this.receiver = receiver;
        this.expression = expression;
    }

    public String toString(){
        return sender + "." + expression + "->" + receiver;
    }

    public String getSender() {
        return sender;
    }

    public String getReceiver() {
        return receiver;
    }

    public BooleanExpression getExpression() {
        return expression;
    }
}

File: pe/src/main/kotlin/ast/cc/nodes/Communication.kt
package ast.cc.nodes

import ast.cc.interfaces.Interaction

data class Communication(override val sender: String, override val receiver: String, val expression: String): Interaction {
    override fun toString(): String {
        val sb = StringBuilder()
        sb.append(sender).append(".").append(expression).append("->").append(receiver).append("; ")
        return sb.toString()
    }
}
